src/codecs/ico/encoder.rs

```diff
@@ -19,6 +20,72 @@ pub struct IcoEncoder<W: Write> {
     w: W,
 }
 
+/// An ICO image entry
+pub struct IcoFrame<'a> {
+    // Pre-encoded PNG or BMP
+    encoded_image: Cow<'a, [u8]>,
+    // Stored as `0 => 256, n => n`
+    width: u8,
```

Comment:
Wait, why `u8`?

Reply:
Because ICOs have a 8bit width and height, so this pre-encodes that instead of doing it late into the encoding stage